lib/std/mem/Allocator.zig

```diff
@@ -6,29 +6,37 @@ const math = std.math;
 const mem = std.mem;
 const Allocator = @This();
 const builtin = @import("builtin");
+const Alignment = std.mem.Alignment;
 
 pub const Error = error{OutOfMemory};
 pub const Log2Align = math.Log2Int(usize);
 
 /// The type erased pointer to the allocator implementation.
-/// Any comparison of this field may result in illegal behavior, since it may be set to
-/// `undefined` in cases where the allocator implementation does not have any associated
-/// state.
+///
+/// Any comparison of this field may result in illegal behavior, since it may
+/// be set to `undefined` in cases where the allocator implementation does not
+/// have any associated state.
 ptr: *anyopaque,
 vtable: *const VTable,
 
 pub const VTable = struct {
-    /// Attempt to allocate exactly `len` bytes aligned to `1 << ptr_align`.
+    /// Allocate exactly `len` bytes aligned to `alignment`, or return `null`
+    /// indicating the allocation failed.
     ///
     /// `ret_addr` is optionally provided as the first return address of the
     /// allocation call stack. If the value is `0` it means no return address
     /// has been provided.
-    alloc: *const fn (ctx: *anyopaque, len: usize, ptr_align: u8, ret_addr: usize) ?[*]u8,
+    ///
+    /// The returned slice of memory must have been `@memset` to `undefined`
+    /// by the allocator implementation.
+    alloc: *const fn (*anyopaque, len: usize, alignment: Alignment, ret_addr: usize) ?[*]u8,
 
-    /// Attempt to expand or shrink memory in place. `buf.len` must equal the
-    /// length requested from the most recent successful call to `alloc` or
-    /// `resize`. `buf_align` must equal the same value that was passed as the
-    /// `ptr_align` parameter to the original `alloc` call.
+    /// Attempt to expand or shrink memory in place.
+    ///
+    /// `memory.len` must equal the length requested from the most recent
+    /// successful call to `alloc` or `resize`. `alignment` must equal the same
```

Comment:
Shouldn't `remap` also be mentioned? Ditto for the other functions with this and similar comments.